src/vs/workbench/browser/parts/views/views.ts

```diff
@@ -678,23 +699,64 @@ export class ViewsService extends Disposable implements IViewsService {
 	}
 
 	async openView(id: string, focus: boolean): Promise<IView | null> {
-		const viewContainer = Registry.as<IViewsRegistry>(ViewExtensions.ViewsRegistry).getViewContainer(id);
+		const viewContainer = this.viewContainers.get(id);
 		if (viewContainer) {
-			const viewletDescriptor = this.viewletService.getViewlet(viewContainer.id);
-			if (viewletDescriptor) {
-				const viewlet = await this.viewletService.openViewlet(viewletDescriptor.id, focus) as IViewsViewlet | null;
-				if (viewlet && viewlet.openView) {
-					return viewlet.openView(id, focus);
+			const location = this.viewContainersRegistry.getViewContainerLocation(viewContainer);
+			const compositeDescriptor = this.getComposite(viewContainer.id, location!);
+			if (compositeDescriptor) {
+				const paneComposite = await this.openComposite(compositeDescriptor.id, location!, focus) as IPaneComposite | undefined;
+				if (paneComposite && paneComposite.openView) {
+					return paneComposite.openView(id, focus);
 				}
 			}
 		}
 
 		return null;
 	}
 
+	moveViews(views: IViewDescriptor[], viewContainer: ViewContainer): void {
+		if (!views.length) {
+			return;
+		}
+
+		const from = this.viewContainers.get(views[0].id);
+		const to = viewContainer;
+
+		if (from && to && from !== to) {
+			this.onDidDeregisterViews(from, views);
+			this.onDidRegisterViews(viewContainer, views);
+			this._onDidChangeContainer.fire({ views, from, to });
+		}
+	}
+
+	getViews(container: ViewContainer): IViewDescriptor[] {
+		return this.viewsRegistry.getViews(container);
+	}
+
+	private async openComposite(compositeId: string, location: ViewContainerLocation, focus?: boolean): Promise<IPaneComposite | undefined> {
+		const panelService = this.instantiationService.invokeFunction(accessor => accessor.get(IPanelService));
```

Comment:
I do not think this is correct usage of services. If it is difficult to extract panel part, how about extracting views service into `IViewDescriptorService` and `IViewsService` ? Former deals only with view and view container descriptors and later deals with views (opening, focusing etc) ?